src/main/java/de/teamlapen/vampirism/player/vampire/VampirePlayer.java

```diff
@@ -216,15 +245,54 @@ public void biteEntity(int entityId) {
             return;
         }
         if (e != null && e instanceof EntityLivingBase) {
-
             if (e.getDistance(player) <= player.getEntityAttribute(EntityPlayer.REACH_DISTANCE).getAttributeValue() + 1) {
-                biteEntity((EntityLivingBase) e);
+                if (determineBiteType(((EntityLivingBase) e)) == BITE_TYPE.ATTACK) {
+                    biteEntity(((EntityLivingBase) e));
+                } else {
+                    victim = e.getEntityId();
+                    PotionEffect effect = new PotionEffect(PotionBloodLoss.POTION, 20, 7);
+                    ((EntityLivingBase) e).addPotionEffect(effect);
+
+                    PotionEffect feedingEffect = new PotionEffect(PotionFeeding.POTION, 25);
+                    player.addPotionEffect(feedingEffect);
+
+                    NBTTagCompound nbt = new NBTTagCompound();
+                    nbt.setInteger(KEY_VICTIM_ID, victim);
+                    sync(nbt, true);
+
+                }
             } else {
                 VampirismMod.log.w(TAG, "Entity sent by client is not in reach " + entityId);
             }
         }
     }
 
+    public void biteVictim() {
+        EntityLivingBase e = ((EntityLivingBase) player.world.getEntityByID(victim));
+        if (e == null) {
+            endBiting();
+            return;
+        }
+        PotionEffect effect = new PotionEffect(PotionBloodLoss.POTION, 20, 7);
```

Comment:
It might look less chaotic if both potion effects do not cause particle effects as they tend to clutter the screen. If you the PotionEffect(Potion,duration,amplifier,ambient,particle) constructor you can disable them